### PR TITLE
added auth.js middleware

### PR DESCRIPTION
**Pull request by divyam234**
@yusukebe  This PR adds middleware for Auth.js to work with hono and vitejs mainly. See if any changes are needed.

### CODE REVIEW

File: packages/next-auth/package.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      "import": {
+        "types": "./dist/react.d.mts",
+        "default": "./dist/react.mjs"
+      },
+      "require": {
+        "types": "./dist/react.d.ts",
+        "default": "./dist/react.js"
+      }
+    }
+  },
```
Comment (yusukebe) [inline packages/next-auth/package.json:27]: We should add the following to make it exporting type correctly:

```json
  "typesVersions": {
    "*": {
      "react": [
        "./dist/react.d.mts"
      ]
    }
  },
```

**Comment by yusukebe** [comment]
Hi @divyam234 

I would like to try this NextAuth.js Middleware, but creating a reproducible application in my environment is difficult. Is this designed to be used within Next.js? If possible, it would be great if you could make a minimum workable example application anywhere!

**Comment by divyam234** [comment]
@yusukebe Will examples folder work fine in same?

**Comment by divyam234** [comment]
@yusukebe  https://github.com/divyam234/next-auth-hono-react modified next auth template anddeployed on cf pages with vite

**Comment by yusukebe** [comment]
@divyam234 

Thanks a lot! It works well and feels good. I'd like to merge it.

One thing I'm considering is the middleware name. Should it be "Auth.js Middleware"?

**Comment by zisra** [comment]
Should be [Auth.js](https://authjs.dev/) not [NextAuth.js](https://next-auth.js.org/)
Screenshot 2023-12-26 at 11 04 01 AM

**Comment by divyam234** [comment]
@yusukebe  pr is ready to be  merge  renamed all components to auth.js.

File: packages/auth-js/README.md [hunk 2]
```diff
@@ -0,0 +1,142 @@
+# Auth js middleware for Hono
+
+This is a [Auth js](https://next-auth.js.org) third-party middleware for [Hono](https://github.com/honojs/hono).
```
Comment (yusukebe) [inline packages/auth-js/README.md:3]: URL should be `https://authjs.dev/` and `Auth js` should be `Auth.js`.

File: packages/auth-js/README.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
+# Auth js middleware for Hono
```
Comment (yusukebe) [inline packages/auth-js/README.md:1]: `Auth js` should be `Auth.js`.

Please fix the other parts as well.

**Review by yusukebe — COMMENTED** [review]
@divyam234 

I found minor issues. Please correct!

File: packages/auth-js/README.md (hunk 3 above)
Comment (bhunter17) [reply packages/auth-js/README.md]: @yusukebe  should be good now

**Comment by divyam234** [comment]
@yusukebe should be good now.

**Review by yusukebe — APPROVED** [review]

**Comment by yusukebe** [comment]
@divyam234 

Thanks! I'll merge and release it now 🎉

**Comment by thakuad** [comment]
Hi, @divyam234 can you please provide an example of how to protect routes, such as /dashboard, so that only authenticated users can access them? I’d like to use Next.js middleware for this.

**Comment by divyam234** [comment]
@thakuad  Its already there in readme also there is a exmaple repo mentioned there.

**Pull request merged**